File: src/main/java/gargoyle/ct/config/convert/impl/CTConfigConverter.java
package gargoyle.ct.config.convert.impl;

import gargoyle.ct.config.CTConfig;
import gargoyle.ct.config.convert.CTUnitConverter;
import org.jetbrains.annotations.NotNull;

import java.util.concurrent.TimeUnit;

public final class CTConfigConverter implements CTUnitConverter<CTConfig> {
    private final CTConfigDataConverter configDataConverter = new CTConfigDataConverter();

    @NotNull
    @Override
    public String format(TimeUnit unit, @NotNull CTConfig data) {
        return configDataConverter.format(TimeUnit.MINUTES, data.getWhole(), data.getBlock(), data.getWarn());
    }

    @NotNull
    @Override
    public CTConfig parse(@NotNull String data) {
        long[] parsed = configDataConverter.parse(data);
        return new CTConfig(parsed[0], parsed[1], parsed[2]);
    }
}

File: src/main/kotlin/gargoyle/ct/config/convert/impl/CTConfigConverter.kt
package gargoyle.ct.config.convert.impl

import gargoyle.ct.config.CTConfig
import gargoyle.ct.config.convert.CTUnitConverter
import java.util.concurrent.TimeUnit

class CTConfigConverter : CTUnitConverter<CTConfig> {
    private val configDataConverter = CTConfigDataConverter()
    override fun format(unit: TimeUnit, data: CTConfig): String {
        return configDataConverter.format(TimeUnit.MINUTES, data.whole, data.block, data.warn)
    }

    override fun parse(data: String): CTConfig {
        val parsed = configDataConverter.parse(data)
        return CTConfig(parsed[0], parsed[1], parsed[2])
    }
}
